File: stack/daily_temperatures/js/daily_temperatures.js
const temps = [73,74,75,71,69,72,76,73];

const dailyTemperatures = (temps) => {
  let answer = new Array(temps.length).fill(0);
  let stack = [];

  for(let i in temps) {
    while(stack.length && temps[i] > stack.at(-1).temp) {
      let top = stack.pop();
      answer[top.iter] = i - top.iter;
    }
    stack.push({ temp: temps[i], iter: i });
  }
  return answer;
};

console.log(dailyTemperatures(temps));

File: stack/daily_temperatures/cpp/daily_temperatures.cpp
#include<iostream>
#include<utility>
#include<vector>
#include<stack>

using std::cout;
using std::vector;
using std::stack;
using std::pair;

// Time O(n)
// Space O(n)

class Solution {
public:
    static vector<int> dailyTemperatures(const vector<int>& temps) {
        vector<int> answer(temps.size(), 0);
        stack<pair<int, int>> stack;

        for(int i = 0; i < temps.size(); i++) {
            while(!stack.empty() && temps[i] > stack.top().first) {
                pair<int, int> top = stack.top();
                answer[top.second] = i - top.second;
                stack.pop();
            }
            stack.push(std::make_pair(temps[i], i));
        }

        return answer;
    }
};

int main() {
    vector<int> temps = { 73, 74, 75, 71, 69, 72, 76, 73 };

    auto answer = Solution::dailyTemperatures(temps);

    for(auto& n : answer) {
        cout << n << ' ';
    }
    cout << '\n';
}
